scripts/utils/esmain.ts

```diff
@@ -0,0 +1,40 @@
+import path from 'path';
```

Comment:
This resource seems like a slightly simpler solution, but maybe you've already tried this and it didn't work? https://2ality.com/2022/07/nodejs-esm-main.html#determining-if-an-esm-module-is-main

Reply:
Yep! That was my first attempt, but this solution doesn't take extensions into account.